### ISSUE TITLE
Could not read typeID: read only 1 of 4 bytes from rabbit_common rabbit_backing_queue.beam

### ISSUE THREAD

**Issue by GrantJamesPowell** [KronicDeth/intellij-elixir#1028]
# Version

7.2.1

# Exception

## Message

Could not read typeID: read only 1 of 4 bytes from #{DIR}/_build/dev/lib/rabbit_common/ebin/rabbit_backing_queue.beam

## Stacktrace

```
java.lang.Throwable: Could not read typeID: read only 1 of 4 bytes from /Users/grantpowell/src/portal/_build/dev/lib/rabbit_common/ebin/rabbit_backing_queue.beam
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:136)
	at org.elixir_lang.beam.chunk.Chunk.typeID(Chunk.java:71)
	at org.elixir_lang.beam.chunk.Chunk.from(Chunk.java:32)
	at org.elixir_lang.beam.Beam.from(Beam.java:109)
	at org.elixir_lang.beam.Beam.lambda$from$0(Beam.java:134)
	at java.util.Optional.flatMap(Optional.java:241)
	at org.elixir_lang.beam.Beam.from(Beam.java:134)
	at org.elixir_lang.beam.psi.BeamFileImpl.buildFileStub(BeamFileImpl.java:90)
	at org.elixir_lang.beam.StubBuilder.buildStubTree(StubBuilder.java:31)
	at com.intellij.psi.stubs.StubTreeBuilder.buildStubTree(StubTreeBuilder.java:61)
	at com.intellij.psi.stubs.StubUpdatingIndex$3.a(StubUpdatingIndex.java:232)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:907)
	at com.intellij.psi.stubs.StubUpdatingIndex$3.map(StubUpdatingIndex.java:214)
	at com.intellij.psi.stubs.StubUpdatingIndex$3.map(StubUpdatingIndex.java:197)
	at com.intellij.util.indexing.impl.MapReduceIndex.mapInput(MapReduceIndex.java:273)
	at com.intellij.util.indexing.VfsAwareMapReduceIndex.calculateUpdateData(VfsAwareMapReduceIndex.java:98)
	at com.intellij.util.indexing.impl.MapReduceIndex.update(MapReduceIndex.java:213)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:1654)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:1582)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.freezeFileTypeTemporarilyIn(FileTypeManagerImpl.java:511)
	at com.intellij.util.indexing.FileBasedIndexImpl.a(FileBasedIndexImpl.java:1541)
	at com.intellij.util.indexing.FileBasedIndexImpl.indexFileContent(FileBasedIndexImpl.java:1529)
	at com.intellij.util.indexing.FileBasedIndexImpl.processRefreshedFile(FileBasedIndexImpl.java:1508)
	at com.intellij.util.indexing.FileBasedIndexProjectHandler.a(FileBasedIndexProjectHandler.java:161)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.a(CacheUpdateRunner.java:227)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1127)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.a(CacheUpdateRunner.java:243)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.project.CacheUpdateRunner$MyRunnable.run(CacheUpdateRunner.java:240)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:315)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by KronicDeth** [comment]
I'll need more information to reproduce.

1. What version of OTP are you using
2. How did you install OTP/Erlang
3. What version of Elixir are you using?
4. How did you install Elixir?
5. What version of `rabbit_common` are you using?

If you have a public repository that reproduces the error, that would be most helpful.

**Comment by sh41** [comment]
Same defect as #1816, fixed in v11.1.0, September 2019.

The message means the file held fewer than four bytes when it was read — it says nothing about the
module in the path. I could only reproduce it by truncating a `.beam` to one, two or three bytes; the
real files decompile fine. A dependency's `ebin` is written by the compiler while the IDE is indexing,
so a read landing mid-write fits.

The log was removed deliberately, in 56b7c0900 and 88cc56783: a partially-written file isn't
actionable, and the plugin picks it up once it is complete.

If a `.beam` still fails to open for you, that is a different bug and worth its own issue.

Closed as part of an AI-assisted review of the backlog. A human checked this one before closing it, but
it might still be a mistake. If you think the reason above is wrong, please reopen the issue and tell us
why.
